src/lib/core/platform/platform.ts

```diff
@@ -0,0 +1,41 @@
+import {Injectable, NgModule, ModuleWithProviders} from '@angular/core';
+
+// Declare window with type of any.
+declare const window: any;
+
+// Whether the current platform supports the V8 Break Iterator. The V8 check
+// is necessary to detect all Blink based browsers.
+const hasV8BreakIterator = (window.Intl && (window.Intl as any).v8BreakIterator);
+
+@Injectable()
+export class MdPlatform {
+
+  /** Layout Engines */
+  EDGE = /(edge)/i.test(navigator.userAgent);
+  TRIDENT = /(msie|trident)/i.test(navigator.userAgent);
+
+  // EdgeHTML and Trident mock Blink specific things and need to excluded from this check.
+  BLINK = !!(window.chrome || hasV8BreakIterator) && !!CSS && !this.EDGE && !this.TRIDENT;
+
+  // Webkit is part of the userAgent in EdgeHTML Blink and Trident, so we need to
+  // ensure that Webkit runs standalone and is not use as another engines base.
+  WEBKIT = /AppleWebKit/i.test(navigator.userAgent) && !this.BLINK && !this.EDGE && !this.TRIDENT;
+
+  /** Browsers and Platform Types */
+  IOS = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream;
+  FIREFOX = /(firefox|minefield)/i.test(navigator.userAgent);
```

Comment:
That's because it would be difficult to detect the plain Gecko layout engine, because actually every browser identifies it self as a Gecko-like browser.

Since we only need to check for Firefox explicit, we can just implement the `firefox` check instead of the unstable `gecko` check (http://webaim.org/blog/user-agent-string-history/)